nomos-services/data-availability/dispersal/src/backend/kzgrs.rs

```diff
@@ -4,19 +4,28 @@ use std::time::Duration;
 // crates
 use futures::StreamExt;
 use itertools::izip;
+use serde::{Deserialize, Serialize};
 // internal
 use crate::adapters::mempool::DaMempoolAdapter;
 use crate::adapters::network::DispersalNetworkAdapter;
 use crate::backend::DispersalBackend;
 use kzgrs_backend::common::blob::DaBlob;
 use kzgrs_backend::common::{build_blob_id, Column, ColumnIndex};
-use kzgrs_backend::encoder::EncodedData;
+use kzgrs_backend::encoder::{DaEncoderParams, EncodedData};
 use kzgrs_backend::{dispersal, encoder};
 use nomos_core::da::{BlobId, DaDispersal, DaEncoder};
 use overwatch_rs::DynError;
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct EncoderSettings {
+    pub num_columns: usize,
+    pub with_cache: bool,
+    pub global_params_path: String,
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct DispersalKZGRSBackendSettings {
-    pub encoder_settings: encoder::DaEncoderParams,
+    pub encoder_settings: EncoderSettings,
```

Comment:
Nitpick: Is this because `DaEncoderParams` does not implement deserialize? In that case I would make the `EncoderSettings` type private and use `remote`:

```rust
#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct DispersalKZGRSBackendSettings {
    #[serde(remote = "EncoderSettings")]
    pub encoder_settings: encoder::DaEncoderParams,
    ...
}
```

Reply:
It's also because `encoder::DaEncoderParams` contains `toeplitz1cache: Option<Toeplitz1Cache>` and `global_parameters: GlobalParameters` which doesn't implement `Debug` and serde.
Using remote we'd still need to implement the `Debug`?

Reply:
Ok, that makes more sense. No then